tcp/client.go

```diff
@@ -0,0 +1,58 @@
+package tcp
+
+import (
+	"encoding/binary"
+	"net"
+
+	"github.com/influxdb/influxdb/data"
+)
+
+type Client struct {
+	conn net.Conn
+}
+
+func NewClient() *Client {
+	return &Client{}
+}
+
+func (c *Client) Dial(addr string) error {
+	conn, err := net.Dial("tcp", addr)
+	if err != nil {
+		return err
+	}
+
+	c.conn = conn
+	return nil
+}
+
+func (c *Client) WriteShardRequest(shardID uint64, points []data.Point) error {
+	mt := writeShardRequestMessage
+	if err := binary.Write(c.conn, binary.LittleEndian, &mt); err != nil {
```

Comment:
This is unnecessary. You should make the request type an enum in the `Request` object. Then change `WriteShardRequest` to just be a `Request` object that has a type in it and then optional other objects contained in it. The first would be a `WriteToShard`. Don't need to duplicate request in the object name since it would always be contained in the request object.